test/univariates.jl

```diff
@@ -75,10 +82,18 @@ function verify_and_test(D::Union{Type,Function}, d::UnivariateDistribution, dct
     # conversions
     if D isa Type && !isconcretetype(D)
         @test convert(D{partype(d)}, d) === d
-        d32 = convert(D{Float32}, d)
-        @test d32 isa D{Float32}
+        if D <: DiscreteUniform #|| D  <: Hypergeometric
+            d32 = convert(D{Int32}, d)
+            @test d32 isa D{Int32}
+            @test partype(d32) == Int32
+        else
+            d32 = convert(D{Float32}, d)
+            @test d32 isa D{Float32}
+            @test partype(d32) == Float32
+        end
```

Comment:
If you remove the type parameter from `DiscreteUniform` again, you should not be able to enter this branch anymore (and hence do not need to handle this case here) since `isconcretetype(DiscreteUniform) === true`.
```suggestion
        d32 = convert(D{Float32}, d)
        @test d32 isa D{Float32}
        @test partype(d32) == Float32
```

Reply:
Modifying DiscreteUniform was intentend to support this test-case of a parametric Integer type. I will move it to its own pull request.